File: Android/tests/src/jp/co/riso/smartdeviceapp/model/printsettings/OptionTest.java
package jp.co.riso.smartdeviceapp.model.printsettings;

import java.io.IOException;
import java.io.StringReader;
import java.util.ArrayList;
import java.util.List;

import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;
import javax.xml.parsers.ParserConfigurationException;

import jp.co.riso.android.util.AppUtils;
import jp.co.riso.smartdeviceapp.SmartDeviceApp;

import org.w3c.dom.Document;
import org.w3c.dom.NodeList;
import org.xml.sax.InputSource;
import org.xml.sax.SAXException;

import android.test.AndroidTestCase;
import android.util.Log;

public class OptionTest extends AndroidTestCase {
    private static final String TAG = "OptionTest";
    private List<Option> mOptionList = new ArrayList<Option>();

    @Override
    protected void setUp() throws Exception {
        super.setUp();
        String xmlString = AppUtils.getFileContentsFromAssets(SmartDeviceApp.getAppContext(), "printsettings.xml");

        Document printSettingsContent = null;

        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
        try {
            DocumentBuilder db = dbf.newDocumentBuilder();

            InputSource is = new InputSource();
            is.setCharacterStream(new StringReader(xmlString));
            printSettingsContent = db.parse(is);
        } catch (ParserConfigurationException e) {
            Log.e(TAG, "Error: " + e.getMessage());
        } catch (SAXException e) {
            Log.e(TAG, "Error: " + e.getMessage());
        } catch (IOException e) {
            Log.e(TAG, "Error: " + e.getMessage());
        }

        if (printSettingsContent == null) {
            return;
        }

        NodeList optionList = printSettingsContent.getElementsByTagName("option");

        // looping through all item nodes <item>
        for (int i = 0; i < optionList.getLength(); i++) {
            Option option = new Option(optionList.item(i));
            mOptionList.add(option);
        }
    }

    @Override
    protected void tearDown() throws Exception {
        super.tearDown();
        mOptionList.clear();
    }

    public void testPreConditions() {
        assertEquals(64, mOptionList.size());
    }

    public void testGetTextContent() {
        assertEquals("ids_lbl_colormode_auto", mOptionList.get(0).getTextContent());
        assertEquals("ids_lbl_colormode_fullcolor", mOptionList.get(1).getTextContent());
        assertEquals("ids_lbl_colormode_black", mOptionList.get(2).getTextContent());
        assertEquals("ids_lbl_orientation_portrait", mOptionList.get(3).getTextContent());
    }
}

File: Android/app/src/androidTest/java/jp/co/riso/smartdeviceapp/model/printsettings/OptionTest.kt
package jp.co.riso.smartdeviceapp.model.printsettings

import android.util.Log
import jp.co.riso.android.util.AppUtils
import jp.co.riso.smartdeviceapp.SmartDeviceApp
import junit.framework.TestCase
import org.junit.After
import org.junit.Before
import org.junit.Test
import org.w3c.dom.Document
import org.xml.sax.InputSource
import org.xml.sax.SAXException
import java.io.IOException
import java.io.StringReader
import javax.xml.parsers.DocumentBuilderFactory
import javax.xml.parsers.ParserConfigurationException

class OptionTest {

    private val _optionList: MutableList<Option> = ArrayList()

    @Before
    fun setUp() {
        val xmlString =
            AppUtils.getFileContentsFromAssets(SmartDeviceApp.getAppContext(), "printsettings.xml")
        var printSettingsContent: Document? = null
        val dbf = DocumentBuilderFactory.newInstance()
        try {
            val db = dbf.newDocumentBuilder()
            val `is` = InputSource()
            `is`.characterStream = StringReader(xmlString)
            printSettingsContent = db.parse(`is`)
        } catch (e: ParserConfigurationException) {
            Log.e(TAG, "Error: " + e.message)
        } catch (e: IOException) {
            Log.e(TAG, "Error: " + e.message)
        } catch (e: SAXException) {
            Log.e(TAG, "Error: " + e.message)
        }
        if (printSettingsContent == null) {
            return
        }
        val optionList = printSettingsContent.getElementsByTagName("option")

        // looping through all item nodes <item>
        for (i in 0 until optionList.length) {
            val option = Option(optionList.item(i))
            _optionList.add(option)
        }
    }

    @After
    fun tearDown() {
        _optionList.clear()
    }

    @Test
    fun testPreConditions() {
        TestCase.assertEquals(345, _optionList.size)
    }

    @Test
    fun testGetTextContent() {
        TestCase.assertEquals("ids_lbl_colormode_auto", _optionList[0].textContent)
        TestCase.assertEquals("ids_lbl_colormode_fullcolor", _optionList[1].textContent)
        TestCase.assertEquals("ids_lbl_colormode_black", _optionList[2].textContent)
        TestCase.assertEquals("ids_lbl_orientation_portrait", _optionList[3].textContent)
    }

    companion object {
        private const val TAG = "OptionTest"
    }
}